ethereum/rlp/src/main/java/org/hyperledger/besu/ethereum/rlp/RLPOutput.java

```diff
@@ -187,7 +188,7 @@ default void writeLong(final long l) {
    *     {@code b < 0} or {@code b > 0xFF}.
    */
   default void writeUnsignedByte(final int b) {
-    writeBytes(Bytes.of(b));
+    processZeroByte(Long.valueOf(b), a -> writeBytes(Bytes.of(b)));
```

Comment:
As b is an integer, could Bytes.of(b) be more than one byte long? With the other writeUnsignedXXX we are making sure that there aren't more bytes than we want (e.g. Bytes.ofUnsignedShort in writeUnsignedShort). Do we have to do something similar here as well?

Reply:
Within Bytes.of() there's a check to confirm whether the value fits within a byte, which throws an IllegalArgumentException if the value would be truncated.